tests/unit/test_charm.py

```diff
@@ -1,35 +1,88 @@
 # Copyright 2020 Ubuntu
 # See LICENSE file for licensing details.
 
+import hashlib
 import random
 import subprocess
 import unittest
+from base64 import b64decode
 from unittest.mock import Mock, call, mock_open, patch
 from uuid import uuid4
 
 from jinja2 import Template
 from ops.model import ActiveStatus, BlockedStatus
 from ops.testing import Harness
 
-from charm import NginxCharm
+from charm import NginxCharm, file_hash, install_ca_cert, write_file
 
-DEFAULT_CONFIG = {"host": str(uuid4()), "port": random.randint(10, 20)}
+DEFAULT_CONFIG = {
+    "host": str(uuid4()),
+    "port": random.randint(10, 20),
+    "ssl_cert": "dGVzdF9jZXJ0==",  # Padded base64 strings
+    "ssl_key": "dGVzdF9rZXk=",  # Padded base64 strings
+    "ca_cert": "dGVzdF9jYV9jZXJ0==",
+}
+NEW_CONFIG = {
+    "host": str(uuid4()),
+    "port": random.randint(10, 20),
+    "ssl_cert": "dGVzdF9jZXJ0==",  # Padded base64 strings
+    "ssl_key": "dGVzdF9rZXk=",  # Padded base64 strings
+    "ca_cert": "dGVzdF9jYV9jZXJ0==",
+}
 STORED_CONFIG = {"host": str(uuid4()), "port": random.randint(10, 20), "publishes": {}}
 
 
 class TestCharm(unittest.TestCase):
-    def test_config_changed(self):
+    @patch("charm.install_ca_cert")
+    @patch("charm.write_file")
+    @patch("os.path.exists", return_value=False)
+    @patch("os.makedirs")
+    @patch("os.chown")
+    @patch("os.chmod")
+    @patch("subprocess.check_call")
+    def test_config_changed(
+        self,
+        mock_check_call,
+        mock_chmod,
+        mock_chown,
+        mock_makedirs,
+        mock_path_exists,
+        mock_write_file,
+        mock_install_ca_cert,
+    ):
         harness = Harness(NginxCharm)
         self.addCleanup(harness.cleanup)
         harness.begin()
         harness.charm._render_config = Mock()
         harness.charm._reload_config = Mock()
-        # default_config = DEFAULT_CONFIG
         harness.update_config(DEFAULT_CONFIG)
-        default_config = {**DEFAULT_CONFIG, "publishes": {}}
+        harness.update_config(NEW_CONFIG)
```

Comment:
question: isn't it redundant to perform two change configs? iiuc only the second update matters